### PR TITLE
dev/migrate-from-vue-moment-to-luxon

### PR DESCRIPTION
**Pull request by PeterThrels**
Replace moment with luxon, nuxt2

### CODE REVIEW

File: components/accounts/statement/statement.vue [hunk 1]
```diff
@@ -118,6 +118,16 @@ export default class AccountStatement extends mixins(
...
 
+    // TODO: Resolve to useBase Composable
+    formatDate(val) {
+        const dateTime = DateTime.fromJSDate(val)
+        if (dateTime.hasSame(DateTime.now(), 'year')) {
+            return dateTime.toFormat('d MMMM')
+        }
+
+        return dateTime.toFormat('d MMMM yyyy')
+    }
```
Comment (oleggrech-t) [inline components/accounts/statement/statement.vue:129]: Yep was gonna mention that maybe this should go to a utils file or base composable as you mentioned

File: components/cards/statement/statement.vue [hunk 2]
```diff
@@ -109,6 +109,15 @@ export default class CardStatement extends mixins(
...
+    formatDate(val) {
+        const dateTime = DateTime.fromJSDate(val)
+        if (dateTime.hasSame(DateTime.now(), 'year')) {
+            return dateTime.toFormat('d MMMM')
+        }
+
+        return dateTime.toFormat('d MMMM yyyy')
+    }
+
     filterMonthChange(val) {
```
Comment (oleggrech-t) [inline components/cards/statement/statement.vue:121]: Can this function also go to the baseComposable or util file?

**Review by garyThrels — APPROVED** [review]
Should you ever need to perform such refactoring, I suggest you move all date and time logic into a single script and have that script use luxon or moment.
Easier to unit test the functions and should you ever need to switch again the code is not split in different scripts

**Review by jasnvella2 — APPROVED** [review]

**Comment by PeterThrels** [comment]
> Should you ever need to perform such refactoring, I suggest you move all date and time logic into a single script and have that script use luxon or moment. Easier to unit test the functions and should you ever need to switch again the code is not split in different scripts

Great suggestion, this will be done eventually.

File: components/accounts/statement/statement.vue (hunk 1 above)
Comment (PeterThrels) [reply components/accounts/statement/statement.vue:129]: correct!

File: components/cards/statement/statement.vue (hunk 2 above)
Comment (PeterThrels) [reply components/cards/statement/statement.vue:121]: this will be done but not in this PR

**Pull request merged**